server/room.ts

```diff
@@ -30,205 +32,16 @@ export class Room {
     }
 
     setInterval(() => {
-      // console.log(roomId, this.video, this.roster, this.tsMap, this.nameMap);
       io.of(roomId).emit('REC:tsMap', this.tsMap);
     }, 1000);
 
     io.of(roomId).on('connection', (socket: Socket) => {
-      // console.log(socket.id);
-      this.roster.push({ id: socket.id });
-
+      new Connection(socket, this);
       socket.emit('REC:host', this.getHostState());
```

Comment:
Is there a reason these aren't in Connection as well? I assume you'd have to change all the references to e.g. this.room.nameMap.

Or this should probably not be in Connection:
```
    this.room.roster.push({ id: socket.id });
    this.room.io.emit('roster', this.room.roster);
```

Those calls and the socket.emit should probably be colocated